### ISSUE TITLE
SagePay Cancel action leads to blank page

### ISSUE THREAD

**Issue by kurund** [eileenmcnaughton/nz.co.fuzion.omnipaymultiprocessor#228]
- Click on 'Confirm' button on contribution page and you will be redirected to SagePay payment screen
- Instead of completing the payment, click on 'Cancel' action
- This displays blank page, https://live.sagepay.com/gateway/service/cancellation
- This might be happening because incorrect response is given back to SagePay
- It would be good to send the use to back to CiviCRM page in such screnarios.

I did some debugging and found that it happens in following function. 
https://github.com/eileenmcnaughton/nz.co.fuzion.omnipaymultiprocessor/blob/b80d86716ec25a57c204e72c9d87ddc083c0d13a/CRM/Core/Payment/OmnipayMultiProcessor.php#L943

$redirectURL is not set for such action.

Potential Solution
- We need to use same logic as in Successful payment to redirect to Thank you or back to Confirm page

**Comment by eileenmcnaughton** [comment]
@kurund it should be set in this line (before the redirect) https://github.com/eileenmcnaughton/nz.co.fuzion.omnipaymultiprocessor/blob/b80d86716ec25a57c204e72c9d87ddc083c0d13a/CRM/Core/Payment/OmnipayMultiProcessor.php#L241

Not sure why it isn't....

**Comment by kurund** [comment]
@eileenmcnaughton 

Yes, but for some reason on error it's no getting the $redirectURL.
